### PR TITLE
[MOBILE-297] provide a way to enable notifications and get the auth result

### PR DESCRIPTION
**Pull request by crow**
### What do these changes do?
These changes provide an interface for the newly added enableUserPushNotifications function in UAPush. This allows users to enable push with a promise that returns the auth result.

### Why are these changes necessary?
Users want to be able to enable push and receive the authorization result back immediately. 

### How did you verify these changes?
Manual testing in the sample.

### CODE REVIEW

**Comment by crow** [comment]
Failing with - /Users/vagrant/git/ios/UARCTModule/UrbanAirshipReactModule.m:71:22: error: no visible @interface for 'UAPush' declares the selector 'enableUserPushNotifications:'. This will go away when the enableUserPushNotifications feature is released in the SDK.

**Comment by marc-scig** [comment]
🐤

File: js/UrbanAirship.js [hunk 1]
```diff
@@ -153,6 +153,16 @@ class UrbanAirship {
     return UrbanAirshipModule.isUserNotificationsEnabled();
   }
 
+  /**
+   * Enables user notifications.
+   *
+   * @return {Promise.<boolean>} A promise that returns true if enablement was authorized
+   * or an false if enablement was rejected
```
Comment (BrianBatchelder) [inline js/UrbanAirship.js]: or ~an~ false

File: js/index.d.ts [hunk 2]
```diff
@@ -110,6 +110,14 @@ export class UrbanAirship {
      */
     static isUserNotificationsEnabled(): Promise<boolean>;
 
+    /**
+     * Enables user push notifications.
+     *
+     * @return {Promise.<boolean>} A promise that returns true if enablement was authorized
+     * or an false if enablement was rejected.
```
Comment (BrianBatchelder) [inline js/index.d.ts]: or ~an~ false

**Comment by rlepinski** [comment]
@crow You need to add the Android version as well. It can just enable notifications and return true.

**Pull request merged**